feg/gateway/sbi/sbi_test.go

```diff
@@ -0,0 +1,158 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sbi_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/labstack/echo/v4"
+	"github.com/stretchr/testify/assert"
+
+	n7_client "magma/feg/gateway/sbi/specs/TS29512NpcfSMPolicyControl"
+	n7 "magma/feg/gateway/sbi/specs/TS29512NpcfSMPolicyControlServer"
+	sbi "magma/feg/gateway/sbi/specs/TS29571CommonData"
+)
+
+type MockPcf struct {
+	ListenAddr string
+	Policies   map[string]n7.SmPolicyControl
+}
+
+func NewMockPcf() *MockPcf {
+	return &MockPcf{
+		Policies: make(map[string]n7.SmPolicyControl),
+	}
+}
+
+func (pcf *MockPcf) PostSmPolicies(ctx echo.Context) error {
+	log.Println("PostSmPolicies")
+	var newPolicy n7.SmPolicyContextData
+	err := ctx.Bind(&newPolicy)
+	if err != nil {
+		return ctx.NoContent(http.StatusBadRequest)
+	}
+
+	var policy n7.SmPolicyControl
+	policy.Context = newPolicy
+
+	pcf.Policies["1234"] = policy
+
+	return ctx.NoContent(http.StatusOK)
+}
+
+// GetSmPoliciesSmPolicyId handles GET /sm-policies/{smPolicyId}
+func (pcf *MockPcf) GetSmPoliciesSmPolicyId(ctx echo.Context, smPolicyId string) error {
+	policy, found := pcf.Policies[smPolicyId]
+	if !found {
+		return ctx.NoContent(http.StatusNotFound)
+	}
+	return ctx.JSON(http.StatusOK, policy)
+}
+
+// PostSmPoliciesSmPolicyIdDelete handles POST /sm-policies/{smPolicyId}/delete
+func (pcf *MockPcf) PostSmPoliciesSmPolicyIdDelete(ctx echo.Context, smPolicyId string) error {
+
+	return ctx.NoContent(http.StatusOK)
+}
+
+// PostSmPoliciesSmPolicyIdUpdate handles POST /sm-policies/{smPolicyId}/update
+func (pcf *MockPcf) PostSmPoliciesSmPolicyIdUpdate(ctx echo.Context, smPolicyId string) error {
+
+	return ctx.NoContent(http.StatusOK)
+}
+
+func TestCreateSMPolicy(t *testing.T) {
+	// Start the server and wait till the server is started
+	mockPcf := startServer(t)
+
+	// Create new N7 client object
+	client, err := n7_client.NewClientWithResponses(fmt.Sprintf("http://%s", mockPcf.ListenAddr))
+
+	assert.NoError(t, err, "Client creation failed")
+
+	// Create a new SM Policy
+	ipAddr := sbi.Ipv4Addr("10.1.2.3")
+	imsi := sbi.Supi("12345678901234")
+	body := n7_client.PostSmPoliciesJSONRequestBody{
+		Ipv4Address:  &ipAddr,
+		PduSessionId: 10,
+		Supi:         imsi,
+	}
+
+	resp, err := client.PostSmPolicies(context.Background(), body)
+	assert.NoError(t, err, "Failed to create SM Policies")
+	assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+	// Get SM Policy given the policyId
+	policyResp, err := client.GetSmPoliciesSmPolicyIdWithResponse(context.Background(), "1234")
+	assert.NoError(t, err, "Failed to get SM Policies")
+	assert.Equal(t, http.StatusOK, policyResp.StatusCode())
+	assert.Equal(t, &ipAddr, policyResp.JSON200.Context.Ipv4Address)
+	assert.Equal(t, imsi, policyResp.JSON200.Context.Supi)
```

Comment:
in my opinion is way better to have something  like this

```
const IMSI1 =  "12345678901234"

assert.Equal(t, sbi.Supi(IMSI1), policyResp.JSON200.Context.Supi)

or 

assert.Equal(t, IMSI1 ,  fmt.Sprint(policyResp.JSON200.Context.Supi)
....


```

Otherwise you have to look in the test for what was imsi.

Same for the other test cases